src/deploy/functions/iambinder.ts

```diff
@@ -0,0 +1,286 @@
+import { createHash } from "crypto";
+
+import * as iam from "../../gcp/iam";
+import * as secretManager from "../../gcp/secretManager";
+import * as resourceManager from "../../gcp/resourceManager";
+import * as backend from "./backend";
+import * as storage from "../../gcp/storage";
+import * as v2Events from "../../functions/events/v2";
+import { assertExhaustive } from "../../functional";
+
+type ResourceManagerResource = `//cloudresourcemanager.googleapis.com/projects/${string}`;
+type SecretManagerResource = `//secretmanager.googleapis.com/projects/${string}/secrets/${string}`;
+type Resource = ResourceManagerResource | SecretManagerResource;
+
+function isResourceManagerResource(r: Resource): r is ResourceManagerResource {
+  return r.startsWith("//cloudresourcemanager.googleapis.com/");
+}
+function isSecreteManagerResource(r: Resource): r is SecretManagerResource {
+  return r.startsWith("//secretmanager.googleapis.com/");
```

Comment:
I feel a bit iffy about this approach. What if IAM binder took the resource address as a parameter?